src/test/unit_tests/test_s3_bucket_policy.js

```diff
@@ -335,6 +337,100 @@ mocha.describe('s3_bucket_policy', function() {
         }));
     });
 
+    mocha.it('should not allow principal get object bucket policy with 2 statements: ' +
```

Comment:
could you think about more tests cases that can cause conflicts? 
I could think of for example - 
1. 2 statements same account - first statement by id = deny, second statement by name = allow
2. 2 statements - first deny all, second - allow account

Reply:
I can try, thanks.

Reply:
I didn't add the "deny all" statement as it would not allow me any operation on it (even the preparation step of adding the object and the policy deletion in the end).
So I added your first idea as two additional tests, thanks.

Reply:
you can add deny all user from a specific operation, not necessary all operations...

Reply:
I can try, thanks.

Reply:
I added your second idea as well, thanks.